### PR TITLE
Support callbacks with keyword arguments

### PR DESCRIPTION
**Pull request by tbennun**
- [x] Escape string arguments such that they generate valid code
- [x] Support kwargs, variable kwargs, and duplicate calls to same function

### CODE REVIEW

File: dace/frontend/python/astutils.py [hunk 1]
```diff
@@ -663,3 +663,11 @@ def create_constant(value: Any, node: Optional[ast.AST] = None) -> ast.AST:
...
 
     return newnode
+
+
+def escape_string(value: str):
+    """ Converts special Python characters in strings back to their parsable version (e.g., newline to ``\n``) """
+    if sys.version_info >= (3, 0):
+        return value.encode("unicode_escape").decode("utf-8")
+    # Python 2.x
+    return value.encode('string_escape')
```
Comment (alexnick83) [inline dace/frontend/python/astutils.py:673]: Isn't this a bit pointless? I think that DaCe is completely incompatible with Python 2.

File: dace/frontend/python/preprocessing.py [hunk 2]
```diff
@@ -300,6 +300,33 @@ def has_replacement(callobj: Callable, parent_object: Optional[Any] = None, node
...
+    Used for creating callbacks from C to Python with keyword arguments.
+    """
+    if not node.keywords:
+        return func
+    keywords = [kw.arg for kw in node.keywords]
+    poscount = len(node.args)
+
+    # Using two levels of functions to ensure keywords are stored with the callback
+    if poscount == 0:
+        def make_cb(keywords, _):
```
Comment (alexnick83) [inline dace/frontend/python/preprocessing.py:315]: Why do we need two different definitions? Some issue with `(*)args`?

**Review by alexnick83 — APPROVED** [review]
Just some questions, nothing blocking

File: dace/frontend/python/astutils.py (hunk 1 above)
Comment (tbennun) [reply dace/frontend/python/astutils.py:673]: True

**Pull request merged**